modules/mqtt/destination/mqtt-worker.c

```diff
@@ -31,7 +31,46 @@
 #define TIMEOUT     10000L
 #define DISCONNECT_TIMEOUT 10000
 
+static LogThreadedResult
+_publish_result_evaluation (LogThreadedDestWorker *self, gint result)
+{
+  switch(result)
+    {
+    case MQTTCLIENT_SUCCESS:
+      return LTR_SUCCESS;
 
+    case MQTTCLIENT_DISCONNECTED:
+      msg_error("Disconnected during publish!",
+                log_pipe_location_tag(&self->owner->super.super.super));
+      return LTR_NOT_CONNECTED;
+
+    case MQTTCLIENT_MAX_MESSAGES_INFLIGHT:
+      msg_error("Max message inflight! (publish)",
+                log_pipe_location_tag(&self->owner->super.super.super));
+      return LTR_ERROR;
+
+    case MQTTCLIENT_FAILURE:
+      msg_error("Failure during publishing!",
+                log_pipe_location_tag(&self->owner->super.super.super));
+      return LTR_ERROR;
+
+    case MQTTCLIENT_PERSISTENCE_ERROR:
+      g_assert_not_reached();
+    case MQTTCLIENT_BAD_QOS:
+      g_assert_not_reached();
+    case MQTTCLIENT_BAD_STRUCTURE:
+      g_assert_not_reached();
+    default:
+      g_assert_not_reached();
+
+    case MQTTCLIENT_NULL_PARAMETER:
+    case MQTTCLIENT_BAD_UTF8_STRING:
+      msg_error("Error during publish!",
```

Comment:
Please be a bit more verbose with this.
Something like: `An unrecoverable error occurred during publish, dropping message.`